### DISCUSSION TITLE
Gngr and Garlic Salata - Add I2P Support

### DISCUSSION THREAD

**Discussion by za3ter2p** [UprootLabs/gngr#279] [Dev Talk]
Hello There,
I was wondering since you guys taking privacy and security into consideration wouldnt be nice adding I2Pjava anonymization to the browser? 

I2P as well has Java based version more: 

https://geti2p.net/en/

I2P Community.

**Comment by eyedeekay** [comment]
I am considering using gngr as a tool to provide a pure-Java browser which is configured to use I2P for platforms where no suitable browser is available using what I call `i2p.plugins.firefox`. If I want to pre-configure gngr, instead of configuring a proxy interactively, is there a recommended way to do that? This would allow me to produce a gngr-based I2P Browsing tool.

**Comment by hrj** [comment]
Thanks for your interest @za3ter2p and @eyedeekay 

In the near future, we don't have the bandwidth to integrate i2p into gngr itself. But supporting it via proxy server should be possible. Hence, I am converting this issue into a discussion.

@eyedeekay Interesting question. The proxy settings are unfortunately being stored in a binary format currently. So there is no clean way to create the settings. You could save your settings and then use that as a template when installing gngr for a new user, via a script. It's ugly, but quick. The connection settings are stored in the file `~/.gngr/default/ConnectionSettings`. The other option is to fork the code and change the default settings, so that the compiled jar file itself has the proxy setting by default.
